test/strscan/test_stringscanner.rb

```diff
@@ -693,6 +723,17 @@ def test_check_until
     assert_equal(nil, s.check_until(/Qux/))
   end
 
+  def test_check_until_string
+    s = create_string_scanner("Foo Bar Baz")
+    assert_equal("Fo", s.check_until("o"))
+    assert_equal(0, s.pos)
```

Comment:
Why do we need this case?
It seems that `s.check_until("Bar")` covers this case.

Reply:
Certainly not necessary.
I will remove this.
Thanks.